spec/integration/api/search_controller_spec.rb

```diff
@@ -460,6 +460,41 @@ def add_contact(*aspects)
       )
       expect(response.status).to eq(401)
     end
+
+    context 'when the author is blocked' do
+      before do
+        FactoryBot.create(
+          :block,
+          user: auth_public_only_read_only.user,
+          person: eve.person
+        )
+        FactoryBot.create(
+          :block,
+          user: auth.user,
+          person: eve.person
+        )
+      end
+
+      it "hides the blocked author's posts in with public only scopes" do
+        get(
+          "/api/v1/search/posts",
+          params: {tag: "tag2", access_token: access_token_public_only_read_only}
+        )
+        expect(response.status).to eq(200)
+        posts = response_body_data(response)
+        expect(posts.length).to eq(1)
```

Comment:
I think it would be a good idea to actually check if the one post that gets returned here is the one that we actually expected, so there can't be a bug where there is only one post returned but it's the one from eve?

But on the other hand, all the other specs here also only check the number of posts, but I also think it would be a good idea to check more there too (for example if only public posts should be returned, check if all the posts that got returned are actually public).